components/locale_core/src/extensions/unicode/value.rs

```diff
@@ -61,16 +62,156 @@ impl Value {
         Ok(Self(v))
     }
 
-    #[doc(hidden)]
+    /// Returns a reference to a single [`Subtag`] if the [`Value`] contains exactly one
+    /// subtag, or `None` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Value;
+    /// use std::str::FromStr;
+    ///
+    /// let value1 = Value::from_str("foo")
+    ///     .expect("failed to parse a Value");
+    /// let value2 = Value::from_str("foo-bar")
+    ///     .expect("failed to parse a Value");
+    ///
+    /// assert!(value1.as_single_subtag().is_some());
+    /// assert!(value2.as_single_subtag().is_none());
+    /// ```
     pub const fn as_single_subtag(&self) -> Option<&Subtag> {
         self.0.single()
     }
 
+    /// Destructs into a single [`Subtag`] if the [`Value`] contains exactly one
+    /// subtag, or returns `None` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Value;
+    /// use std::str::FromStr;
+    ///
+    /// let value1 = Value::from_str("foo")
+    ///     .expect("failed to parse a Value");
+    /// let value2 = Value::from_str("foo-bar")
+    ///     .expect("failed to parse a Value");
+    ///
+    /// assert!(value1.into_single_subtag().is_some());
+    /// assert!(value2.into_single_subtag().is_none());
+    /// ```
+    pub fn into_single_subtag(self) -> Option<Subtag> {
+        self.0.into_single()
+    }
```

Comment:
Discussion: I don't see a good reason to have a moving `into_single_subtag` function since `Subtag` is `Copy`. Seems like `as_single_subtag(&self)` would be fine.

Reply:
The value is that you don't need to copy even - you destruct into a single Subtag. Let me know if that convinces you, I'll replace it with `as_single_subtag().copied()` otherwise.

Reply:
Maybe just make `as_single_subtag` return `Option<Subtag>`.

Reply:
Optional: rename the function `to_single_subtag`.

Reply:
as per https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

```
as_ -> Free       ->  borrowed -> borrowed
to_ -> Expensive  ->  borrowed -> borrowed
                      borrowed -> owned (non-Copy types)
                      owned -> owned (Copy types)
into_ -> Variable ->  owned -> owned (non-Copy types)
```

We have:
* `as_single_subtag(&self) -> Option<&Subtag> `
* `into_single_subtag(self) -> Option<Subtag>`

I think `as_` and `into_` fit - `to_` could work if source was Copy, but it's not, so passing it has to consume (`to_` should take reference).

Reply:
Well it's arguable that it is working on Copy types because it returns Some if the thing is a single Subtag (Copy). But I don't feel strongly about this and it need not block the PR

Reply:
I checked that. The closest reference in Stdlib I could find it `Cow` which has:
* `to_mut(&mut self) -> &mut <B as ToOwned>::Owned`
* `into_owned(self) -> <B as ToOwned>::Owned` (vs. OSstring::to_owned(&mut self) -> T`.

The latter has also the following description:
> Creates owned data from borrowed data, usually by cloning.

So my read is that `to_` is meant to be expensive, it takes a borrowed value. Our case is cheap, it takes an owned value.

I agree that it's not a clear cut, but I don't take `&self` and I don't clone. `to_X(self) -> T` is only for primitive types such as `f64::to_bits`.

I'll keep `into_*`.